modules/client-configuration/pages/contact-methods-saltbundle.adoc

```diff
@@ -0,0 +1,81 @@
+Using the Salt bundle, it is no longer needed that the client provides the required Python version or any Python modules.
+
+If you bootstrap new clients, registration with the Salt bundle is the default method.
+You can switch existing clients to the Salt bundle method.
+If you switch, the [package]``salt-minion`` package and its Python version and the Python modules will stay installed.
+
+=== Switching from Salt Minion to Salt Bundle
+
+The Salt state [literal]``util.mgr_switch_to_venv_minion`` is available to switch from [package]``salt-minion`` to [package]``venv-salt-minion``.
+It is recommended to switch to [package]``venv-salt-minion`` in two steps to avoid trouble with shifting processes:
+
+.Procedure: Switching with [literal]``util.mgr_switch_to_venv_minion`` state to [package]``venv-salt-minion``
+
+. Apply [literal]``util.mgr_switch_to_venv_minion`` with no pillar specified first.
+  This will result in the switch to [package]``venv-salt-minion`` with copying configuration files etc.
+  It will not clean up the original [package]``salt-minion`` configurations and its packages.
++
+----
+salt <minion_id> state.apply util.mgr_switch_to_venv_minion
+----
+. Apply [literal]``util.mgr_switch_to_venv_minion`` with [literal]``mgr_purge_non_venv_salt`` set to [literal]``True`` to remove [package]``salt-minion`` and with [literal]``mgr_purge_non_venv_salt_files`` set to [literal]``True`` to remove all the files related to [package]``salt-minion``.
+This second step repeats the first step, and then removes the old configuration files and the now obsolete [package]``salt-minion`` package.
++
+----
+salt <minion_id> state.apply util.mgr_switch_to_venv_minion pillar='{"mgr_purge_non_venv_salt_files": True, "mgr_purge_non_venv_salt": True}'
+----
+
+On the other hand, it is also possible to avoid installing the Salt bundle and keep using [package]``salt-minion`` instead.
+In this case specify one of these options:
+
+* Execute [command]``mgr-bootstrap`` with  [option]``--no-bundle`` option.
+* Set [literal]``AVOID_VENV_SALT_MINION`` to [literal]``1`` in the generated bootstrap script.
+* For bootstrap state set the [literal]``mgr_avoid_venv_salt_minion`` pillar to  [literal]``True``.
+
+== Salt SSH with the Salt Bundle
+
+The Salt Bundle is also used when performing Salt SSH actions to clients.
+
+A pre flight script is executed as part of the Salt SSH execution which deploys the code from the Salt Bundle on the target client without installing the package. The original `salt-thin` method is not used in this case.
+
+Salt API is calling Python 3 deployed with the Bundle to handle Salt SSH session on the client, so Salt SSH (including bootstrapping with {webui}) is not dependant on Python installed on the system.
+
+[IMPORTANT]
+====
+The bootstrap repository must be created before bootstrapping the client with {webui}. The pre flight script is detecting the OS on the client and using the bootstrap repository relevant for the OS version.
```

Comment:
I think we don't need to expose the "pre flight script" name to our users, I see it only as an implementation detail.
```suggestion
The bootstrap repository must be created before bootstrapping the client with {webui}. The Salt Bundle used by Salt SSH is taken from the bootstrap repository based on the detected target operating system.
```

Reply:
Not sure if it sounds more clear in passive than before.

Reply:
I'm also in favor of not mentioning pre flight script, maybe we can put the sentence in a different way but without mentioning pre flight script.